src/services/StreamService.ts

```diff
@@ -0,0 +1,65 @@
+import {DynamoDBRecord} from "aws-lambda";
+import {DynamoDB} from "aws-sdk";
+import {cloneDeep} from "lodash";
+import {Utils} from "../utils/Utils";
+import {ITechRecordWrapper, Plates} from "../utils";
+
+/**
+ * Service class for interpreting and formatting
+ * incoming DynamoDB streams
+ */
+class StreamService {
+
+    /**
+     * Extract MODIFY events from the DynamoDB Stream, convert them
+     * to a JS object and expand the tech records into multiple ones for each plate
+     * Example:
+     * Convert
+     * tech-record
+     *  ├── new-plate-1
+     *  ├── new-plate-2
+     *  into
+     *  tech-record
+     *  └── new-plate-1
+     *  tech-record
+     *  └── new-plate-2
+     * @param event
+     */
+    public static getTechRecordsStream(event: any) {
+        // Create from a tech record with multiple plates, multiple tech records with one plate each
+        console.log("Filtering only MODIFY events");
+        const records: any[] = event.Records.filter((record: DynamoDBRecord) => { // Retrieve "MODIFY" events
+            return record.eventName === "MODIFY";
+        })
+            .map((record: DynamoDBRecord) => { // Convert to JS object
+                if (record.dynamodb && record.dynamodb.NewImage) {
+                    return DynamoDB.Converter.unmarshall(record.dynamodb.NewImage);
+                }
+            });
+        console.log("UNMARSHALLED EVENTS", JSON.stringify(records));
```

Comment:
Is this log necessary?

Reply:
I added it to see how the event looks like and create some test data. I will remove most of these logs after we test the entire flow